### DISCUSSION TITLE
TanStackRouterVite Configuration with tsr.config.js

### DISCUSSION THREAD

**Discussion by BrendanC23** [TanStack/router#1333] [Q&A]
I'm using the `TanStackRouterVite()` and I can configure it by doing the following:

```ts
plugins: [
    react(),
    TanStackRouterVite({
        routesDirectory: "my-path/routes",
        generatedRouteTree: "my-path/routeTree.gen.ts",
        routeFileIgnorePrefix: "-",
        quoteStyle: "double",
    }),
]
```

However, when I move the TSR config to a `tsr.config.json` file, the Vite plugin does not detect the configuration file. Though the CLI `tsr generate` command does read the `tsr.config.json`.

Is there a straightforward way to share configuration information so that it can be read by both the CLI and the Vite plugin?

**Comment by SeanCassiere** [comment]
Is your `tsr.config.json` file not at the root of your project? because by default we read the `.json` configuration file before touching the plugin configuration.
